lib/NormalModule.js

```diff
@@ -327,6 +330,50 @@ class NormalModule extends Module {
 		};
 		const loaderContext = {
 			version: 2,
+			getOptions: (loaderName, schema) => {
+				const loader = loaderContext.loaders[loaderContext.loaderIndex];
+				let options = {};
+
+				if (loader.options && typeof loader.options === "object") {
+					({ options } = loader);
+				} else if (loader.query) {
+					let { query } = loader;
+
+					if (query.substr(0, 1) !== "?") {
+						throw new WebpackError(
+							"A valid query string should begin with '?'"
+						);
+					}
+
+					query = query.substr(1);
+
+					// Allow to use `?foo=bar` in `options`
+					if (query.substr(0, 1) === "?") {
+						query = query.substr(1);
+					}
+
+					if (query.substr(0, 1) === "{" && query.substr(-1) === "}") {
+						try {
+							options = parseJson(query);
+						} catch (e) {
+							throw new WebpackError(`Cannot parse query string: ${e.message}`);
+						}
+					} else {
+						options = querystring.parse(query);
+					}
+				}
+
+				if (!schema) {
+					return options;
+				}
+
+				validateOptions(schema, options, {
+					name: loaderName || "Unknown Loader",
```

Comment:
I think its fine to use `getCurrentLoaderName()` here and omit the `loaderName` argument.